### PR TITLE
Update requirements to require scikit-learn and purge old requirements

### LINKED ISSUE #119: Pytables dependency removal
Source code inspection for the conda-forge feedstock found that pytables is not referenced anywhere and could be removed as a dependency. Is this true, and if so should it be removed from both the conda-forge and local requirements?

### PR DESCRIPTION
**Pull request by freemansw1**
Resolves #119 and #196. 

**Specific actions**
* Removes `pytables` and `cf-units` from our dependencies (both are used by our dependencies, but not by us directly)
* Renames `conda-requirements.txt` to `requirements.txt` as that seems to be the standard (I cannot find reference to `conda-requirements.txt` anywhere)
* Adds requirements to `setup.py` such that `pip install .` installs dependencies as well. 

Standard questions
* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [ ] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [ ] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [ ] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [ ] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Comment by freemansw1** [comment]
Note that the requirements in `setup.py` do not match those in `requirements.txt` because cartopy builds through `pip` appear to be broken, and our `black` runner depends on `pip install .` working.

**Review by snilsn — CHANGES_REQUESTED** [review]
Good job @freemansw1, makes sense to me. There is a minor issue that I have found.

The README.md still references the old name of requirements.txt

`conda install -c conda-forge --yes --file conda-requirements.txt`

**Review by snilsn — APPROVED** [review]
Great! Approving this now

File: setup.py [hunk 1]
```diff
@@ -49,6 +49,15 @@ def get_version(pkg_name):
...
+    install_requires=[
+        "numpy",
+        "scipy",
+        "scikit-image",
+        "scikit-learn",
+        "pandas",
+        "matplotlib",
+        "xarray",
+        "trackpy",
+    ],
```
Comment (JuliaKukulies) [inline setup.py:61]: What about `iris` and `cartopy` here? 

And should we maybe add `pytest` as `test_requires` ?

**Review by JuliaKukulies — COMMENTED** [review]
Great update @freemansw1!

File: setup.py (hunk 1 above)
Comment (freemansw1) [reply setup.py:61]: See above comment about iris and cartopy. I'm not sure that we can do much until/unless cartopy fixes their pip builds. 

Good catch with `test_requires`, I'll add that shortly.

**Comment by JuliaKukulies** [comment]
All right, approved! 

And sorry that I missed your very clear comment on cartopy and iris! You are probably right that we can not do much about that.

**Comment by freemansw1** [comment]
> All right, approved!
> 
> And sorry that I missed your very clear comment on cartopy and iris! You are probably right that we can not do much about that.

No worries! It is a bit odd, and I do worry that this will cause us headaches when putting things onto PyPI (#192), but it's the solution that we have for now.

**Comment by freemansw1** [comment]
I had to resolve a merge conflict with #191 merged in. @JuliaKukulies or @snilsn can you give this a quick look-over again and then I will merge?

**Comment by snilsn** [comment]
Looks good and ready to be merged @freemansw1!

**Comment by freemansw1** [comment]
Thanks!

**Pull request merged**